debug/linking_utils/ldd.py

```diff
@@ -29,31 +38,75 @@ def f(y):
         return x
     return f
 
+def memoized(cache, f, arg):
```

Comment:
FYI there is, in python3, `functool.lru_cache`.

Reply:
I saw that, but I want to be compatible with 2 and 3